tutorials/scripting/c_sharp/c_sharp_exports.rst

```diff
@@ -0,0 +1,354 @@
+.. _doc_c_sharp_exports:
+
+C# Exports
+==========
+
+Introduction to exports
+-----------------------
+
+In Godot, class members can be exported. This means their value gets saved along
+with the resource (such as the :ref:`scene <class_PackedScene>`) they're
+attached to. They will also be available for editing in the property editor.
+Exporting is done by using the ``[Export]`` attribute::
+
+    public class ExportExample : Spatial
+    {
+        [Export]
+        public int Number = 5;
+    }
+    
+In that example the value ``5`` will be saved, and after building the current project
+it will be visible in the property editor. This way, artists and game designers
+can modify values that later influence how the program runs. For this, a
+special export syntax is provided.
+
+Exporting can only be done with built-in types or objects derived from the :ref:`Resource class <class_Resource>`.
+
+.. note::
+
+    Exporting properties can also be done in GDScript, for information on that
+    see :ref:`doc_gdscript_exports`.
+
+Basic use
+---------
+
+Exporting can work with and without setting a default value. For int and float
+``0`` will then be used as the default.
+
+::
+
+    [Export]
+	public int Number;
+
+Export works with resource types.
+
+::
+
+	[Export]
+	private Texture CharacterFace;
+	[Export]
+	private PackedScene SceneFile;
+
+There are many resource types that can be used this way, try e.g.
+the following to list them:
+
+::
+
+	[Export]
+	private Resource Resource;
+
+.. 
+	Commenting out enum examples becuase I have been told they
+	require extra steps to actually work properly. The examples bellow
+	will show up in the inspector but apparently not function properly
+	..
+	Integers and strings hint enumerated values.
+	..
+	::
+	..
+		// Editor will enumerate as 0, 1 and 2.
+		[Export(PropertyHint.Enum, "Warrior,Magician,Thief")]
+		private int CharacterClass;
+	..
+	If type is String, editor will enumerate with string names.
+	..
+	::
+	..
+		[Export(PropertyHint.Enum, "Rebecca,Mary,Leah")]
+		private string CharacterName;
+	..
+	Named enum values
+	-----------------
+	..
+	Editor will enumerate as THING_1, THING_2, ANOTHER_THING.
+	..
+	::
+	..
+		private enum NamedEnum
+		{
+			Thing1,
+			Thing2,
+			AnotherThing = -1
+		}
+		[Export(PropertyHint.Enum)]
+		private NamedEnum X;
+
+Strings as paths
+----------------
+
+Property hints can be used to export strings as paths
+
+String as a path to a file.
+
+::
+
+    [Export(PropertyHint.File)]
+	private string F;
```

Comment:
One letter names is a bad practice the docs shouldn't teach, imho. There's some more later on.